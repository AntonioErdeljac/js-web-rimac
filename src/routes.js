import { Help, Home, Settings } from './views';

export default [
  {
    path: '/',
    component: Home,
    exact: true,
    icon: 'tachometer-alt',
  },
  {
    path: '/settings',
    component: Settings,
    exact: true,
    icon: 'cog',
  },
  {
    path: '/help',
    component: Help,
    exact: true,
    icon: 'question',
  },
];
