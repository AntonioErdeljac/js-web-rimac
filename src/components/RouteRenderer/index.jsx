import React, { memo } from 'react';
import { Switch, Route } from 'react-router-dom';

import routes from '../../routes';

const RouteRenderer = () => {
  return (
    <Switch>
      {routes.map((route) => (
        <Route key={route.path} exact={route.exact} component={route.component} path={route.path} />
      ))}
    </Switch>
  );
};

export default memo(RouteRenderer);
