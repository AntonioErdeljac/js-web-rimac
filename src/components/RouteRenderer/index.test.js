import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import RouteRenderer from './index';

import routes from '../../routes';

jest.mock('../../routes', () => [
  {
    path: '/lorem',
    name: 'Lorem',
    icon: 'lorem',
    exact: true,
  },
  {
    path: '/ipsum',
    name: 'Ipsum',
    icon: 'Ipsum',
    exact: false,
  },
]);

const renderComponent = () => shallow(<RouteRenderer />);

describe('RouteRenderer Component', () => {
  it('renders correctly according to snapshot', () => {
    expect(toJson(renderComponent())).toMatchSnapshot();
  });

  it('renders proper number of routes', () => {
    const wrapper = renderComponent();

    expect(wrapper.find('Route').length).toBe(routes.length);
  });
});
