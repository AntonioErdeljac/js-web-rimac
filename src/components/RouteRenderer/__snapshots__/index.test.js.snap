// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteRenderer Component renders correctly according to snapshot 1`] = `
<Switch>
  <Route
    exact={true}
    key="/lorem"
    path="/lorem"
  />
  <Route
    exact={false}
    key="/ipsum"
    path="/ipsum"
  />
</Switch>
`;
